pkg/dockerutil/container_host.go

```
@@ -3,30 +3,95 @@ package dockerutil
 import (
 	"fmt"
 	"net"
+	"os/exec"
 	"runtime"
+	"strings"
 
 	"gopkg.in/inconshreveable/log15.v2"
 )
 
+func ifaceIPAddr(ifaceName string) (ip string, err error) {
+	iface, err := net.InterfaceByName(ifaceName)
+	if err != nil {
+		return "", err
+	}
+	addrs, err := iface.Addrs()
+	if err != nil {
+		return "", err
+	}
+	for _, addr := range addrs {
+		if ipn, ok := addr.(*net.IPNet); ok {
+			return ipn.IP.String(), nil
+		}
+	}
+	return "", nil
```

I guess I meant to ask why you don't want to return an error when no named IP was found.